src/main/scala/clump/Clump.scala

```diff
@@ -5,7 +5,7 @@ import com.twitter.util.Throw
 import com.twitter.util.Try
 import com.twitter.util.Return
 
-trait Clump[+T] {
+sealed trait Clump[+T] {
```

Comment:
This means that the trait can't be extended outside of the compilation unit (Clump.scala).